### PR TITLE
mkfifo: better handle the mode + umask

### PR DESCRIPTION
**Pull request by sylvestre**
Should make tests/misc/mknod.sh pass

### CODE REVIEW

**Comment by sylvestre** [comment]
I think I regressed the these two tests

File: tests/by-util/test_mkfifo.rs
Comment (cakebaker) [inline tests/by-util/test_mkfifo.rs:1]: When running the tests with `cargo test --features=mkfifo --no-default-features` I get two "unresolved import" errors:
```
error[E0432]: unresolved import `uucore::fs`
  --> tests/by-util/test_mkfifo.rs:59:17
   |
59 |     use uucore::fs::display_permissions;
   |                 ^^ could not find `fs` in `uucore`
   |
note: found an item that was configured out
  --> /home/dho/projects/coreutils/src/uucore/src/lib/lib.rs:52:26
   |
52 | pub use crate::features::fs;
   |                          ^^
note: the item is gated behind the `fs` feature
  --> /home/dho/projects/coreutils/src/uucore/src/lib/lib.rs:51:7
   |
51 | #[cfg(feature = "fs")]
```
I think you have to add the `fs` feature to `src/uu/mkfifo/Cargo.toml`.

File: tests/by-util/test_mkfifo.rs [hunk 1]
```diff
@@ -52,3 +52,52 @@ fn test_create_one_fifo_already_exists() {
         .fails()
         .stderr_is("mkfifo: cannot create fifo 'abcdef': File exists\n");
 }
+
+#[test]
+#[cfg(unix)]
```
Comment (cakebaker) [inline tests/by-util/test_mkfifo.rs:57]: I guess this is not necessary because it's a unix-only program.

File: tests/by-util/test_mkfifo.rs [hunk 2]
```diff
@@ -52,3 +52,52 @@ fn test_create_one_fifo_already_exists() {
...
+        let permissions = display_permissions(&metadata, true);
+        assert_eq!(permissions, expected.to_string());
+    };
+
+    test_fifo_creation("734", 0o077, "prwx-wxr--"); // spell-checker:disable-line
+    test_fifo_creation("706", 0o777, "prwx---rw-"); // spell-checker:disable-line
+}
+
+#[test]
+#[cfg(unix)]
```
Comment (cakebaker) [inline tests/by-util/test_mkfifo.rs:82]: See my other comment.

**Comment by cakebaker** [comment]
> Congrats! The gnu test tests/misc/mknod is no longer failing!

Cool :)

**Pull request merged**